pkg/aws/appmesh.go

```diff
@@ -1487,3 +1517,34 @@ func durationToMillis(d *appmesh.Duration) *int64 {
 
 	return nil
 }
+
+func convertClientPolicy(c *appmeshv1beta1.ClientPolicy) *appmesh.ClientPolicy {
+	if c == nil {
+		return nil
+	}
+	clientPolicy := &appmesh.ClientPolicy{}
+	if c.Tls != nil {
+		tls := appmesh.ClientPolicyTls{
```

Comment:
Could we be consistent on usage of pointers vs. values? Here we store `clientPolicy` as by pointer and  `tls` within it by value and reference the pointer later. I think storing the pointer is preferred.

Reply:
I was struggling with this a bit. I think by-value is actually preferred from a safety (and performance) perspective? Most of these things are just getting passed around immutably once created.

Though given the nullability in the SDK models, that ends up forcing use of pointers for stuff like the `SetTls` call, or the returned pointer to the `clientPolicy` to the caller.

Could use another opinion.

Reply:
I think small, short-lived structs are preferred to be stored by value, unless a pointer is needed.  Since in this case, the pointer is needed for the SetTls call, I think whether or not you use a pointer up front is a judgement call based on what you think is more readable.

Reply:
Got it. I'll do a pass over and try to keep consistent to using values until a pointer is needed.

Reply:
Makes sense, let’s keep it as a value by now. This can easily be changed later.